crates/mun_syntax/src/ast/expr_extensions.rs

```diff
@@ -174,6 +176,103 @@ impl Literal {
             _ => unreachable!(),
         }
     }
+
+    pub fn text_and_suffix(&self) -> (SmolStr, Option<SmolStr>) {
+        let token = self.token();
+        let text = token.text();
+        match self.kind() {
+            LiteralKind::String => (text.clone(), None),
+            LiteralKind::IntNumber => {
+                let (str, suffix) = split_int_text_and_suffix(text);
+                (SmolStr::new(str), suffix.map(SmolStr::new))
+            }
+            LiteralKind::FloatNumber => {
+                let (str, suffix) = split_float_text_and_suffix(text);
+                (SmolStr::new(str), suffix.map(SmolStr::new))
+            }
+            LiteralKind::Bool => (text.clone(), None),
+        }
+    }
+}
+
+/// Given a string containing an integer literal (e.g `0x123` or `1234u32`), split the string in the
+/// value part and the suffix part.
+fn split_int_text_and_suffix(text: &str) -> (&str, Option<&str>) {
+    let base = match text.as_bytes() {
+        [b'0', b'x', ..] => 16,
+        [b'0', b'b', ..] => 8,
+        [b'0', b'o', ..] => 2,
+        _ => 10,
+    };
+
+    let mut iter = text.char_indices().peekable();
+
+    // Skip base specifier
+    if base != 10 {
+        iter.next();
+        iter.next();
+    }
+
+    // Skip digits in the string
+    skip_digits(base, &mut iter);
+
+    if let Some((idx, _)) = iter.next() {
+        (&text[0..idx], Some(&text[idx..]))
+    } else {
+        (text, None)
+    }
+}
+
+/// Skip all digits in the iterator that belong to the given base
```

Comment:
It's all looking good, so I've been reduced to grammar nitpicking. The subject is 'it', so 'it skips', etc. E.g. in [Rust](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.new).

```suggestion
/// Skips all digits in the iterator that belong to the given base
```